apps/sms/views/shared/test/unit/activity_handler_test.js

```diff
@@ -455,38 +452,38 @@ suite('ActivityHandler', function() {
     });
 
     suite('Close notification', function() {
-      var closeSpy;
-
       setup(function() {
-        closeSpy = this.sinon.spy(Notification.prototype, 'close');
-        this.sinon.stub(document, 'addEventListener');
+        this.sinon.spy(NotificationHelper, 'send');
       });
 
       test('thread view already visible', function() {
         isDocumentHidden = false;
         this.sinon.stub(Threads, 'currentId', message.threadId);
         MockNavigatormozApps.mTriggerLastRequestSuccess();
-        sinon.assert.notCalled(document.addEventListener);
-        sinon.assert.notCalled(closeSpy);
+
+        sinon.assert.notCalled(NotificationHelper.send);
       });
 
       test('Not in target thread view', function() {
         isDocumentHidden = true;
         MockNavigatormozApps.mTriggerLastRequestSuccess();
-        sinon.assert.notCalled(document.addEventListener);
-        sinon.assert.notCalled(closeSpy);
+
+        sinon.assert.notCalled(NotificationHelper.send);
       });
 
       test('In target thread view and view is hidden', function(done) {
-        isDocumentHidden = true;
+        /*isDocumentHidden = true;
         this.sinon.stub(Threads, 'currentId', message.threadId);
 
-        MockNavigatormozApps.mTriggerLastRequestSuccess().then(() => {
-          sinon.assert.called(document.addEventListener);
-          sinon.assert.notCalled(closeSpy);
-          document.addEventListener.yield();
-          sinon.assert.called(closeSpy);
-        }).then(done, done);
+        MockNavigatormozApps.mTriggerLastRequestSuccess().then(
+          () => NotificationHelper.send.firstCall.returnValue
+        ).then((notification) => {
+          sinon.assert.called(notification.addEventListener);
```

Comment:
same here: we want to assert that `document.addEventListener.withArgs('visibilitychange')` is called... But you can just remove this line, as this is implied by the yield we have later on.